boards/riscv32/hifive1/doc/hifive1.rst

```diff
@@ -0,0 +1,60 @@
+.. _hifive1:
+
+SiFive HiFive1
+##############
+
+Overview
+********
+
+The HiFive1 is a development board with the FE310 RISC-V SoC.
+The board is Arduino compatible.
+More information can be found on `SiFive's wesite <https://www.sifive.com/products/hifive1>`_.
+
+Programming and debugging
+*************************
+
+Building
+========
+
+Applications for the ``HiFive1`` board configuration can be built as usual (see :ref:`build_an_application`).
+In order to build the application for ``HiFive1``, set the ``BOARD`` variable to ``hifive1``.
+
+.. code-block:: bash
+
+	$ export ZEPHYR_GCC_VARIANT="zephyr"
+	$ export ZEPHYR_SDK_INSTALL_DIR="/opt/zephyr-sdk" #assuming default sdk install directory
+	$ export BOARD="hifive1"
+
+Flashing
+========
+
+In order to upload the application to the device, you'll need OpenOCD and GDB with RISC-V support.
+Those are easiest obtainable from SiFive's Freedom-E-SDK. Download and installation instructions can be found in the `SDK's GitHub repository <https://github.com/sifive/freedom-e-sdk>`_.
+
+With the necessary tools installed, you can connect to the board using OpenOCD. To establish an OpenOCD connection, switch to the :file:`utils` directory and run:
+
+.. code-block:: bash
+
+	# assuming that the location of the openocd is in PATH
+	$ sudo openocd -f ${SDK_PATH}/bsp/env/freedom-e300-hifive1/openocd.cfg
+
+
+Leave it running, and in a different terminal, use GDB to upload the binary to the board. Use the RISC-V GDB from SiFive's Freedom-E-SDK toolchain.
+
+Before loading, the device's flash protection has to be disabled. In order to load the binary to the device, run the following commands in the GDB terminal:
```

Comment:
Change to:

    Before loading the binary image to the device, the device's flash protection
    must be disabled. Here are the GDB terminal commands to connect to the device,
    disable the flash protection, load the binary, and start it running: